Utils/Dataflow/test/utils/reconstruction.py

```diff
@@ -0,0 +1,99 @@
+"""
+Task chain reconstruction from elasticsearch.
+"""
+import copy
+import sys
+
+import elasticsearch
+
+
+INDEX = 'tasks_chain_data'
+
+
+def es_connect():
+    ''' Establish a connection to elasticsearch.
+
+    TODO: take parameters from es config.
+    '''
+    return elasticsearch.Elasticsearch(['localhost:9200'])
+
+
+def get_chain_id(es, index, taskid):
+    ''' Get chain_id by given taskid. '''
+    srch = {'query': {'match': {'_id': taskid}}}
+    results = es.search(index=index, body=srch)
+    results = results['hits']['hits']
+    if not results:
+        return False
+    else:
+        return results[0]['_source'].get('chain_id')
+
+
+def get_chain_data(es, index, chain_id):
+    ''' Get chain_data of all tasks that have given chain_id. '''
+    srch = {'query': {'match': {'chain_id': chain_id}}}
+    rtrn = []
+    fr = 0
+    sz = 100
+    while True:
+        results = es.search(index=index, body=srch, from_=fr, size=sz)
+        if not results['hits']['hits']:
+            break
+        for hit in results['hits']['hits']:
+            chain_data = hit['_source'].get('chain_data')
+            if chain_data:
+                rtrn.append(chain_data)
+        fr += sz
+    return rtrn
+
+
+def construct_chain(data):
+    ''' Build a chain from the list of its tasks' chain_data. '''
```

Comment:
The function looks fine (I mean, does what it is supposed to), yet it definitely lacks the description of input / output data.
The output structure is a bit complicated to construct and read... wouldn\`t something like nested dicts make it easier?

Reply:
Added description.
Current version was approved by Mikhail in email, over the nested dicts that were suggested as well.